presto-main/src/main/java/com/facebook/presto/execution/QueryStats.java

```diff
@@ -73,6 +76,8 @@
     private final DataSize outputDataSize;
     private final long outputPositions;
 
+    private final List<OperatorStats> operatorSummaries;
```

Comment:
Yes, every distinct operator will get a stats object. Given that we're reducing the refresh interval for stats this might not be too bad. The only other alternative is to introduce special handling for TableFinish earlier in the process?